sdk/core/azure-core/azure/core/pipeline/policies/_retry.py

```diff
@@ -325,7 +326,30 @@ def increment(self, settings, response=None, error=None):
                         )
                     )
 
-        return not self.is_exhausted(settings)
+        if self.is_exhausted(settings):
+            return False
+
+        if response.http_request.body and hasattr(response.http_request.body, 'read'):
+            if 'body_position' not in settings:
+                return False
+            try:
+                # attempt to rewind the body to the initial position
+                response.http_request.body.seek(settings['body_position'], SEEK_SET)
+            except (UnsupportedOperation, ValueError, AttributeError):
+                # if body is not seekable, then retry would not work
+                return False
+        if response.http_request.files and 'file_positions' in settings:
+            try:
+                for name, position in settings['file_positions'].items():
+                    for key, value in response.http_request.files.items():
+                        file_name = value[0]
+                        if file_name == name:
+                            body = value[1]
+                            body.seek(position, SEEK_SET)
```

Comment:
I think this would avoid the double loop:
```
for key, (file_name, body) in response.http_request.files.items():
    if file_name in settings['file_positions']:
        position = settings['file_positions'][file_name]
        body.seek(position, SEEK_SET)
```

Reply:
the value is not always a tuple of 2, sometimes it is a tuple of 3. key, (file_name, body) seems not work?

Reply:
Probably wouldn't hurt to factor out the settings dict access too:
```
file_positions = settings.get('file_positions')
if response.http_request.files and file_positions:
try:
    for key, (file_name, body) in response.http_request.files.items():
        if file_name in file_positions:
            position = file_positions[file_name]
            body.seek(position, SEEK_SET)
except (UnsupportedOperation, ValueError, AttributeError):
    return False
````

Reply:
This is a real scenario:
{
'fileContent': ('tmpfile.txt', <_io.BufferedReader name='tmpfile.txt'>, 'application/octet-stream'),
'fileName': (None, 'tmpfile')
}
We want to save the position for fileContent and skip fileName (not return false)

Reply:
Ah, did not know that. Will have to unpack explicit then. I think the double loop can still be avoided?
```
file_positions = settings.get('file_positions')
if response.http_request.files and file_positions:
    try:
        for key, value in response.http_request.files.items():
            file_name, body = value[0], value[1]
            if file_name in file_positions:
                position = file_positions[file_name]
                body.seek(position, SEEK_SET)
    except (UnsupportedOperation, ValueError, AttributeError):
        return False
```